### PR TITLE
Update `TypeScript Version`s to be at least as high as dependencies' versions

### PR DESCRIPTION
**Pull request by ghost**
Generated automatically.
For each package A depending on a package B, ensures that A has a TypeScript version at least as high as that of B.

Note: this also removed path mappings to "q/v0" from many packages. This was there due to a bug in types-publisher that considered `q/v0` to have a dependency itself, thus the path mapping was considered necessary (it isn't as of #21051).

### CODE REVIEW

File: types/dagre-d3/index.d.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Project: https://github.com/cpettitt/dagre-d3
 // Definitions by: Mark Wong Siang Kai <https://github.com/markwongsk>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
```
Comment (mhegazy) [inline types/dagre-d3/index.d.ts]: this depends on d3, which you just updated to `2.3` so should not this be `2.3` as well?

**Comment by RyanCavanaugh** [comment]
@andy-ms Please fix the failures indicated in the Travis CI log.

**Comment by nycdotnet** [comment]
Probably should note that this field is now significant in the readme.md file.  Possibly under https://github.com/DefinitelyTyped/DefinitelyTyped#edit-an-existing-package section?

**Comment by andy-ms** [comment]
The field has always been significant -- it determines the lowest version the package is tested for. See https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-use-features-from-typescript-21-or-above

**Comment by andy-ms** [comment]
@mhegazy @RyanCavanaugh good to go?

**Pull request merged**